### PR TITLE
feat(runtime): agent workspace model — isolated multi-agent config

### LINKED ISSUE #1093: Phase 7.1: Agent workspace model (isolated config, memory, tools)
### Overview
Define the agent workspace model for multi-agent routing. Each workspace is an isolated agent configuration with its own workspace files (AGENT.md, SOUL.md, etc.), skills, LLM config, memory namespace, capabilities, session config, and tool permissions. Workspaces enable running multiple agent personalities (work, personal, DeFi) under a single Gateway.

### Depends on
- #1053 Phase 1.1: Gateway core process
- #1066 Phase 3.5: Workspace files
- #1083 Phase 5.6: Agent personality templates

### Affected files
- `runtime/src/gateway/workspace.ts` (new)
- `runtime/src/gateway/workspace.test.ts` (new)
- `runtime/src/gateway/index.ts` (modify)

### Types and interfaces

```typescript
export interface AgentWorkspace {
  readonly id: string;
  readonly name: string;
  readonly path: string;
  readonly files: WorkspaceFiles;
  readonly skills: readonly string[];
  readonly llm?: Partial<LLMConfig>;
  readonly memoryNamespace: string;
  readonly capabilities: bigint;
  readonly session?: Partial<SessionConfig>;
  readonly toolPermissions?: readonly ToolPolicy[];
}

export class WorkspaceManager {
  constructor(basePath: string);
  load(workspaceId: string): Promise<AgentWorkspace>;
  listWorkspaces(): Promise<readonly string[]>;
  createWorkspace(id: string, template?: PersonalityTemplate): Promise<AgentWorkspace>;
  deleteWorkspace(id: string): Promise<boolean>;
  getDefault(): string;
}
```

### Test cases
1. `load` reads workspace from directory
2. `listWorkspaces` returns all workspace IDs
3. `createWorkspace` scaffolds directory with template
4. `deleteWorkspace` removes workspace directory
5. `getDefault` returns default workspace ID
6. Workspace has isolated memory namespace
7. Workspace loads workspace files correctly
8. Missing workspace directory throws
9. Invalid workspace config handled gracefully
10. Multiple workspaces coexist

### Acceptance criteria
- [ ] `AgentWorkspace` type with isolated config, memory, tools
- [ ] `WorkspaceManager` with CRUD operations
- [ ] Memory namespace isolation per workspace
- [ ] Template-based workspace creation
- [ ] All new code has test coverage
- [ ] Types exported from barrel `index.ts`

### Scope discipline
- Workspace model and management only
- Do NOT implement routing (Phase 7.2) or sub-agents (Phase 7.4)

### Rollback plan
- Delete workspace module, revert exports

### PR DESCRIPTION
**Pull request by 7etsuo**
## Summary

Implements Phase 7.1: Agent Workspace Model to support multi-agent routing with isolated configurations.

### Key Changes

- **WorkspaceManager**: Provides CRUD operations over workspace directories
- **AgentWorkspace**: Type definition for isolated agent configurations including:
  - Personality files (system prompts, behavior templates)
  - Skills and tool permissions
  - LLM configuration
  - Memory namespace isolation
  - Capabilities and session config
  - Tool policy controls
- **Validation**: WorkspaceValidationError class and validation helpers
- **Exports**: Gateway barrel exports for workspace types and classes

### Implementation Details

Created `runtime/src/gateway/workspace.ts` with:
- `WorkspaceManager.load()`: Load workspace from directory
- `WorkspaceManager.listWorkspaces()`: Enumerate available workspaces
- `WorkspaceManager.createWorkspace()`: Create new workspace from template
- `WorkspaceManager.deleteWorkspace()`: Remove workspace directory
- `WorkspaceManager.getDefault()`: Get default workspace config

### Test Coverage

17 test cases covering:
- Workspace validation (name, structure, config schema)
- CRUD operations (load, list, create, delete)
- Error handling (missing files, invalid configs, duplicate names)
- Default workspace behavior
- Tool policy and personality template validation

All tests passing.

Closes #1093

### CODE REVIEW

**Review by 7etsuo — COMMENTED** [review]
## Review Summary

Overall clean implementation that follows existing gateway module patterns well. 7 items below as inline comments — 3 required fixes, 4 recommended improvements.

**Required:**
1. `createWorkspace` silently overwrites existing workspace config (bug)
2. Hardcoded `fileMap` duplicates `WORKSPACE_FILES` constant (fragile)
3. Missing test coverage for directory-exists-but-no-config edge case

**Recommended:**
4. `deleteWorkspace` should verify workspace.json exists before `rm -rf`
5. `listWorkspaces` skips `MAX_WORKSPACE_ID_LENGTH` check (inconsistency)
6. Capabilities double-parsed in validation + resolve (dead code)
7. `ws` dependency addition in yarn.lock appears unrelated

See inline comments for details.

**Review by 7etsuo — COMMENTED** [review]
Good implementation overall — follows existing gateway patterns, solid test coverage, clean types. 7 items flagged below (3 required, 4 recommended).

File: runtime/src/gateway/workspace.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+    const [config, files] = await Promise.all([
+      loadWorkspaceConfig(workspacePath),
+      new WorkspaceLoader(workspacePath).load(),
+    ]);
+
+    return resolveAgentWorkspace(workspaceId, workspacePath, config, files);
+  }
+
+  async listWorkspaces(): Promise<string[]> {
```
[the remark this answers is not in the document]
Comment (7etsuo) [reply runtime/src/gateway/workspace.ts]: Fixed in 1311176 — added existence check before mkdir. createWorkspace now throws `WorkspaceValidationError('id', 'Workspace already exists: ...')` if the directory already exists. Added corresponding test.

File: runtime/src/gateway/workspace.ts [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+
+  return errors;
+}
+
+// ============================================================================
+// Internal helpers
+// ============================================================================
+
+async function loadWorkspaceConfig(workspacePath: string): Promise<WorkspaceConfigJson> {
+  const configPath = join(workspacePath, WORKSPACE_CONFIG_FILE);
```
[the remark this answers is not in the document]
Comment (7etsuo) [reply runtime/src/gateway/workspace.ts]: Fixed in 1311176 — simplified to direct `BigInt()` call since `validateWorkspaceConfig` already guarantees the string is valid.

File: runtime/src/gateway/workspace.ts [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+export class WorkspaceManager {
+  readonly basePath: string;
+
+  constructor(basePath: string) {
+    this.basePath = basePath;
+  }
+
+  async load(workspaceId: string): Promise<AgentWorkspace> {
+    validateWorkspaceId(workspaceId);
+
```
[the remark this answers is not in the document]
Comment (7etsuo) [reply runtime/src/gateway/workspace.ts]: Fixed in 1311176 — added `entry.length > MAX_WORKSPACE_ID_LENGTH` check to the filter.

**Comment by 7etsuo** [comment]
All 4 HIGH items addressed in 3958675:

**H1** — `WorkspaceValidationError` moved to `gateway/errors.ts` alongside the other gateway error classes. `workspace.ts` now imports from `./errors.js` and re-exports for backwards compatibility. Barrel exports updated — `WorkspaceValidationError` now exported from the errors block.

**H2** — `createWorkspace` existence check now explicitly checks for `ENOENT` and re-throws any other error (e.g. `EACCES`). Same pattern used in `loadWorkspaceConfig` and `listWorkspaces`.

**H3** — Added `session` validation: `if (raw.session !== undefined && !isRecord(raw.session)) errors.push('session must be an object')`. Test added.

**H4** — Added `memoryNamespace` validation: rejects non-string and empty string values. Test added.

21 tests passing, no new type errors.

**Review by 7etsuo — COMMENTED** [review]
## Re-Review — All Issues Fixed

All 4 required changes addressed:
- **H1** FIXED: `WorkspaceValidationError` moved to `gateway/errors.ts`
- **H2** FIXED: `createWorkspace` now properly re-throws non-ENOENT errors
- **H3** FIXED: `session` config validated with `isRecord()` check
- **H4** FIXED: `memoryNamespace` validated as non-empty string

New tests added for each fix. No new issues introduced. LGTM.
